src/gui/ElevatorGUI.java

```diff
@@ -0,0 +1,266 @@
+import java.awt.event.ActionEvent;
+import javax.swing.JTextArea;
+import javax.swing.Timer;
+
+import main.Timing;
+
+/**
+ * The GUI that represents the different elevators that are running.
+ *
+ */
+public class ElevatorGUI extends JFrame {
+
+	private JPanel contentPane, elevatorOutput;
+	// Create a file chooser
+	final JFileChooser fc = new JFileChooser();
+	public FloorSubsystem floorSubsystem;
+	public File file;
+
+	private JCheckBox debugMode;
+	private JTextArea elevatorData0, elevatorData1, elevatorData2, elevatorData3;
+
+	private Elevator e0, e1, e2, e3;
+	private JButton btnStart;
+
+	/**
+	 * Launch the application.
+	 */
+	public static void main(String[] args) {
+		EventQueue.invokeLater(new Runnable() {
+			public void run() {
+				try {
+					ElevatorGUI frame = new ElevatorGUI();
+					frame.setVisible(true);
+				} catch (Exception e) {
+					e.printStackTrace();
+				}
+			}
+		});
+	}
+
+	/**
+	 * Create the frame.
+	 */
+	public ElevatorGUI() {
+		// Get the screen size of the computer
+		Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+		int frameWidth = (int) screenSize.getWidth();
+		int frameHeight = (int) screenSize.getHeight() / 2 + 100;
+
+		int x = (int) (screenSize.getWidth() - frameWidth) / 2;
+		int y = (int) (screenSize.getHeight() - frameHeight) / 2;
+
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setBounds(x, y, frameWidth, frameHeight);
+		contentPane = new JPanel();
+		contentPane.setBorder(new EmptyBorder(5, 5, 5, 5));
+		setContentPane(contentPane);
+
+		JButton btnNewButton = new JButton("Import Input File");
+		btnNewButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				fc.setCurrentDirectory(new File(System.getProperty("user.dir")));
+				int returnVal = fc.showOpenDialog(ElevatorGUI.this);
+				if (returnVal == JFileChooser.APPROVE_OPTION) {
+					file = fc.getSelectedFile();
+				} else {
+				}
+			}
+		});
+
+		addOutputConsoles();
+
+		elevatorFinished(this);
+
+		Configurations.NUM_ELEVATORS = Integer
+				.parseInt(JOptionPane.showInputDialog(this, "Enter the number of Elevators: "));
+		Configurations.NUMBER_OF_FLOORS = Integer
+				.parseInt(JOptionPane.showInputDialog(this, "Enter the number of Floors: "));
+
+		btnStart = new JButton("Start");
+		btnStart.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				floorSubsystem = new FloorSubsystem(file.getName(), Configurations.FLOOR_PORT,
+						Configurations.FLOOR_EVENT_PORT);
+				RunElevator(floorSubsystem);
+			}
+		});
+		contentPane.add(btnStart);
+		contentPane.add(btnNewButton);
+		contentPane.add(elevatorOutput);
+	}
+
+	/**
+	* Produces a pop up whenever the system is finished with an input file
+	*/
+	private void elevatorFinished(JFrame frame) {
+		Timer checkIfFinished = new Timer(5000, new ActionListener() {
+			public void actionPerformed(ActionEvent evt) {
+				String timingInfo = Timing.getTimingInfo();
```

Comment:
is this 5000 ms? II think that's a little short for waiting for input7.txt for ex to run